### PR TITLE
Fix: Fix response streams in template

### PR DESCRIPTION
**Pull request by oscar60310**
### What's happened

DataSources returned streams instead of arrays, but we haven't adjusted the template output:
```sql
{% req user %}
select * from users where id = {{ context.params.id }}
{% endreq %}
{{ user.value()[0].name }} --- this will be undefined
```
This PR consume the stream and download data for templates.

### CODE REVIEW

File: packages/core/src/lib/template-engine/built-in-extensions/query-builder/executorRunner.ts [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
-    const builder: IDataQueryBuilder = value;
-    return builder.value();
+    const response = await builder.value();
+
+    // if input value is not a query builder, call the function .value and do nothing.
+    if (!isDataResult(response)) return response;
+
+    const { getData } = response;
+    const dataStream = getData();
+    const data = await streamToArray(dataStream);
```
Comment (kokokuo) [inline packages/core/src/lib/template-engine/built-in-extensions/query-builder/executorRunner.ts:28]: If we execute `builder.value()` and consume the stream to the array in the `ExecutorRunner` when play `RuntimeExtension` and it's the final result, then how could we convert the array result to json / csv format in the middleware?

File: packages/core/src/lib/template-engine/built-in-extensions/query-builder/executorBuilder.ts [hunk 2]
```diff
@@ -2,9 +2,36 @@ import {
...
+    ) {
+      const args = new nunjucks.nodes.NodeList(node.lineno, node.colno);
+      args.addChild(node.name.target);
+      const filter = new nunjucks.nodes.Filter(
+        node.lineno,
+        node.colno,
+        new nunjucks.nodes.Symbol(node.lineno, node.colno, EXECUTE_FILTER_NAME),
+        args
+      );
+      replace(filter);
```
Comment (kokokuo) [inline packages/core/src/lib/template-engine/built-in-extensions/query-builder/executorBuilder.ts:34]: Curious about `replace`, when we call `replace` what do we replace to ?

**Review by kokokuo — APPROVED** [review]
Beside some question, others LGTM 👍

File: packages/core/src/lib/template-engine/built-in-extensions/query-builder/executorBuilder.ts (hunk 2 above)
Comment (oscar60310) [reply packages/core/src/lib/template-engine/built-in-extensions/query-builder/executorBuilder.ts:34]: It is a [callback function](https://github.com/Canner/vulcan/blob/develop/packages/core/src/lib/template-engine/extension-utils/helpers.ts#L29) to remove or replace the current node in order to modify the AST tree, which provided by visitChildren helper.

**Comment by oscar60310** [comment]
Hi @kokokuo , all issues have been resolve, I also found another bug and fix it, https://github.com/Canner/vulcan/pull/42/commits/4a2fdca4d9325feb2e7a510ed6e9cb40ba3acc86

File: packages/core/src/lib/template-engine/built-in-extensions/query-builder/executorBuilder.ts (hunk 2 above)
Comment (kokokuo) [reply packages/core/src/lib/template-engine/built-in-extensions/query-builder/executorBuilder.ts:34]: Thanks a lot for replying to me

File: packages/core/src/lib/template-engine/built-in-extensions/query-builder/executorRunner.ts (hunk 1 above)
Comment (kokokuo) [reply packages/core/src/lib/template-engine/built-in-extensions/query-builder/executorRunner.ts:28]: Okay, thanks

**Review by kokokuo — APPROVED** [review]
LGTM 👍  cc @oscar60310

**Pull request merged**